openapi/paths/api@courses.yaml

```diff
@@ -0,0 +1,29 @@
+get:
+  operationId: searchCourses
+  tags: [course]
+  summary: Get a list of courses
+  description: |
+    This allows you to search for courses get details about the course. The search filter attributes are all optional.
+    If the limit query parameter is not provided, the number of courses returned is defaulted to 100.
+    This endpoint is only available to superusers and staff
+
+  responses:
+    '200':
+      $ref: ../components/responses/course-list.yaml
+    default:
+      $ref: ../components/responses/problem.yaml
+  parameters:
+    - name: query
+      in: query
+      description: |
+        Will do a fuzzy match on the `course` name
+      schema:
+        type: string
+    - name: folder
+      in: query
+      description: |
+        Passing folder ID will return related courses as suggestions
```

Comment:
We probably need to add some notes as to this is mutually exclusive with the query param.

Along with the logic of how this functions.
This folder contains a curric code, and will show suggestions as to the course that fulfills that curriculum